src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -1159,3 +1159,19 @@ def validator_delete_retention_days(namespace):
         if namespace.delete_retention_days > 365:
             raise ValueError(
                 "incorrect usage: '--delete-retention-days' must be less than or equal to 365")
+
+
+def validate_private_endpoint_connection_id(cmd, namespace):
+
+    if namespace.connection_id:
+        id_parts = namespace.connection_id.split('/')
+        namespace.private_endpoint_connection_name = id_parts[-1]
+        namespace.account_name = id_parts[-3]
+        namespace.resource_group_name = id_parts[-7]
```

Comment:
If the input `connection_id` is illegal, what happens if the array is out of bounds?

Reply:
removed. utilize common validator in core.